__tests__/integration/utils/wallet-helper.js

```diff
@@ -146,18 +155,42 @@ export class WalletHelper {
       walletsArr.unshift(new WalletHelper(genesis.walletId, { words: genesis.words }));
     }
 
-    // Requests the start of all the wallets in quick succession
-    const startPromisesArray = [];
-    for (const wallet of walletsArr) {
-      const promise = TestUtils.startWallet(wallet.walletData);
-      walletsPendingReady[wallet.walletId] = wallet;
-      startPromisesArray.push(promise);
+    // Start of the requests
+    startBenchmark.requestsStart = Date.now().valueOf();
+
+    if (options.serial || walletsArr.length > 2) {
+      // If we need to initialize too many wallets at once, it's better to do it serially
+      for (const wallet of walletsArr) {
+        const walletBenchmark = {};
+        walletBenchmark.requestStart = Date.now().valueOf();
+        await TestUtils.startWallet(wallet.walletData, {
+          waitWalletReady: true
+        });
+        walletBenchmark.requestEnd = Date.now().valueOf();
+        walletsPendingReady[wallet.walletId] = wallet;
+        walletBenchmark.diffRequest = walletBenchmark.requestEnd
+          - walletBenchmark.requestStart;
+        startBenchmark.wallets[wallet.walletId] = walletBenchmark;
+      }
+    } else {
+      // Requests the start of all the wallets in quick succession - parallel mode
+      const startPromisesArray = [];
+      for (const wallet of walletsArr) {
+        const promise = TestUtils.startWallet(wallet.walletData);
+        walletsPendingReady[wallet.walletId] = wallet;
+        const walletBenchmark = {};
+        walletBenchmark.requestStart = Date.now().valueOf();
+        startBenchmark.wallets[wallet.walletId] = walletBenchmark;
+        startPromisesArray.push(promise);
+      }
+      await Promise.all(startPromisesArray);
     }
-    await Promise.all(startPromisesArray);
+
+    startBenchmark.requestsEnd = Date.now().valueOf();
```

Comment:
Maybe `requestsEnd` is not the best name of variable here. You will still do requests to check wallet status.

Reply:
On [7c199cd](https://github.com/HathorNetwork/hathor-wallet-headless/pull/164/commits/7c199cdfff0a5892ea114cd2726dd2ce37b90318) I changed the variable names and added some comments to better explain the process.

In short, there are three phases: we have the `/start` requests phase, the `/status` loop to double-check if the wallets were actually started and the final time calculation.